src/Microsoft.Extensions.Primitives/InplaceStringBuilder.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+
+namespace Microsoft.Extensions.Primitives
+{
+    [DebuggerDisplay("Value = {_value}")]
+    public struct InplaceStringBuilder
```

Comment:
Maybe a lil late, but we should add a comment that this should only be used for well-known reasonably sized strings.  For everything else, use `StringBuilder`. (Primarily because user code would see this via HttpAbstractions \ Mvc)